services/src/test/java/org/exoplatform/onlyoffice/OnlyofficeEditorServiceTest.java

```diff
@@ -254,42 +354,77 @@ public void testCanEditDocument() throws Exception {
   }
   */
 
-  protected void startSessionAs(String user) throws Exception {
-    HashSet<MembershipEntry> memberships = new HashSet<MembershipEntry>();
-    memberships.add(new MembershipEntry("/platform/administrators"));
-    Identity identity = new Identity(user, memberships);
-    ConversationState state = new ConversationState(identity);
-    state.setAttribute(ConversationState.SUBJECT, identity.getSubject());
-    ConversationState.setCurrent(state);
-    SessionProvider provider = new SessionProvider(state);
-    sessionProviderService.setSessionProvider(null, provider);
-    session = provider.getSession(WORKSPACE_NAME, repositoryService.getCurrentRepository());
-  }
 
-  protected NodeImpl createDocument(String title, String type, String data) throws Exception {
-    NodeImpl rootNode = (NodeImpl) session.getRootNode();
-    rootNode.setPermission("john", new String[] { PermissionType.SET_PROPERTY });
-    NodeImpl node = (NodeImpl) rootNode.addNode(title, type);
-    node.addMixin("mix:lockable");
-    node.addMixin("mix:referenceable");
-    node.addMixin("exo:privilegeable");
-    node.addMixin("exo:datetime");
-    node.addMixin("exo:modify");
-    node.addMixin("exo:sortable");
-    node.setProperty("exo:lastModifier", "john");
-    node.setProperty("exo:lastModifiedDate", Calendar.getInstance());
-    if (type.equals("nt:file")) {
-      Node contentNode = node.addNode("jcr:content", "nt:unstructured");
-      contentNode.addMixin("exo:datetime");
-      contentNode.setProperty("jcr:mimeType", "application/vnd.oasis.opendocument.text");
-      contentNode.setProperty("jcr:lastModified", Calendar.getInstance());
-      contentNode.setProperty("jcr:data", data);
-      contentNode.setProperty("exo:dateCreated", Calendar.getInstance());
-      contentNode.setProperty("exo:dateModified", Calendar.getInstance());
+    /**
+     * create document for not root node
+     */
+    protected NodeImpl createDocumentNotRootNode(String title, String type, String data) throws Exception {
```

Comment:
We should change the existing method `createDocument` by addiing a new argument for the parent path instead of creating a new method IMO (which is almost a duplicate).